cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -651,7 +650,6 @@ void SparsePoseGraph::TrimmingHandle::MarkSubmapAsTrimmed(
 
   // Mark the 'nodes_to_remove' as trimmed and remove their data.
   for (const mapping::NodeId& node_id : nodes_to_remove) {
-    CHECK(!parent_->trajectory_nodes_.at(node_id).trimmed());
     parent_->trajectory_nodes_.at(node_id).constant_data.reset();
```

Comment:
You also have to change this code then to use `Trim()` instead of resetting the `constant_data`, right?

Reply:
Done.